protocol/messenger_communities.go

```diff
@@ -2692,7 +2702,9 @@ func (m *Messenger) requestCommunitiesFromMailserver(communities []communities.C
 	m.requestedCommunitiesLock.Lock()
 	defer m.requestedCommunitiesLock.Unlock()
 
-	var topics []types.TopicType
+	// we group topics by PubsubTopic
+	groupedTopics := map[string][]types.TopicType{}
```

Comment:
This could allow duplicated TopicTypes. If that's okay, feel free to ignore, otherwise, i'd suggest to maybedefine the groupedTopics like this:
```
groupedTopics := map[string]map[types.TopicType]struct{}
```

and that way, instead of doing an append like  line 2736, you'd do it like this:
```
groupedTopics[filter.PubsubTopic][filter.ContentTopic] = struct{}{}
```

and then in line 2757, you'd obtain the contentTopics with
```
import "golang.org/x/exp/maps"

maps.Keys(contentTopics)
```

Reply:
Nice idea, thank you! 👍